pkg/catalogv2/helmop/operation.go

```diff
@@ -663,10 +671,68 @@ func (s *Operations) createOperation(ctx context.Context, user user.Info, status
 		return nil, err
 	}
 
+	if err := s.createRoleAndRoleBindings(op, user.GetName()); err != nil {
+		return nil, err
+	}
+
 	op.Status = status
 	return s.ops.UpdateStatus(op)
 }
 
+func (s *Operations) createRoleAndRoleBindings(op *catalog.Operation, user string) error {
+	ownerRef := metav1.OwnerReference{
+		APIVersion: op.APIVersion,
+		Kind:       op.Kind,
+		Name:       op.Name,
+		UID:        op.UID,
+	}
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            name.SafeConcatName(op.GetName(), user, "role"),
+			Namespace:       op.Namespace,
+			OwnerReferences: []metav1.OwnerReference{ownerRef},
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				Verbs:         []string{"get"},
+				Resources:     []string{"operations"},
+				APIGroups:     []string{"catalog.cattle.io"},
+				ResourceNames: []string{op.Name},
+			},
+		},
+	}
+
+	roleBinding := &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            name.SafeConcatName(op.GetName(), user, "rolebinding"),
+			Namespace:       op.Namespace,
+			OwnerReferences: []metav1.OwnerReference{ownerRef},
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				APIGroup: rbacv1.GroupName,
+				Kind:     rbacv1.UserKind,
+				Name:     user,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: rbacv1.GroupName,
+			Kind:     "Role",
+			Name:     role.Name,
+		},
+	}
+
+	if _, err := s.roles.Create(role); err != nil {
```

Comment:
We should check the error for already exists so we don't return it and cause a loop of errors. Same applies below

Reply:
https://github.com/rancher/rancher/blob/19e44aaf2046594f90dbaef8c1883e3d024666c9/pkg/catalogv2/helmop/operation.go#L802 The name is always generated so it is guaranteed to be unique. If it is not unique we should really fail though.